### DISCUSSION TITLE
Create a testcontainer on a remote daemon using SSH, is this supported?

### DISCUSSION THREAD

**Discussion by preetmyob** [testcontainers/testcontainers-dotnet#857] [Q&A] [answered]
*I'm on a mac m1, but this is more of a general question.  I use colima to run containers on arm and x86 platforms, I also connect to other docker daemons *

I can run testcontainers with the Colima docker.sock using the symlinks from `/var/run/docker.sock` 

```
/var/run/docker.sock -> /Users/Preet.Sangha/.docker/run/docker.sock
/Users/Preet.Sangha/.docker/run/docker.sock -> /Users/Preet.Sangha/.colima/edit/docker.sock
```

When using the docker cli,  I can connect to docker daemon running on another (virtual) machine by setting up an appropriate `docker context` using ssh.

```bash
❯ docker context ls --format table                                                                                                                               
NAME               DESCRIPTION                               DOCKER ENDPOINT                                       ERROR
colima-edit *      colima [profile=edit]                     unix:///Users/<user_id>/.colima/edit/docker.sock
default            Current DOCKER_HOST based configuration   unix:///var/run/docker.sock
parallels-debian   parllels-debian                           ssh://parallels@ip.ad.re.ss
``` 

 I've not found a way to connect testcontainers to the remote daemon. Is there a defined way I should set it up, please?

**Comment by HofmeisterAn** [comment]
To use Colima usually you need to set the `DOCKER_HOST` environment variable to Colima's daemon socket. I do not recommend to change the symlink. Please note, Testcontainers requires a Docker API compatible runtime. There might be features Colima does not support.

Regarding SSH. Docker.DotNet does not support SSH endpoints (https://github.com/dotnet/Docker.DotNet/pull/575). Therefore Testcontainers cannot support it.

**Reply by preetmyob** [reply]
Thank you @HofmeisterAn . I've read through that chain of PRs and can see it's predicated on [SSH.NET changes](https://github.com/sshnet/SSH.NET/pull/1052). I look forward to [Docker.DotNet/pull/575 ](https://github.com/dotnet/Docker.DotNet/pull/575)  going into master.

**Reply by preetmyob** [reply]
Do you have any recommended ways of connecting to remote docker daemons instead of SSH at this point? I'll try TCP port forwarding to see how I get on in the mean time

**Reply by HofmeisterAn** [reply, marked as answer]
You can connect to a TCP endpoint, simply set the `DOCKER_HOST` environment variable:

    DOCKER_HOST=tcp://127.0.0.1:2375 dotnet test $PROJECT

**Comment by preetmyob** [comment]
Thank you  this works great
